internal/pipeline/pipeline_test.go

```diff
@@ -620,3 +619,87 @@ func TestValidatePath(t *testing.T) {
 		}
 	}
 }
+
+func TestFunctionConfig(t *testing.T) {
+	type input struct {
+		name              string
+		fn                Function
+		configFileContent string
+		expected          string
+	}
+
+	cases := []input{
+		{
+			name: "no config",
+			fn:   Function{},
+			expected: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: function-input
+data: {}
+`,
+		},
+		{
+			name: "inline config",
+			fn: Function{
+				Config: *yaml.MustParse(`apiVersion: cft.dev/v1alpha1
+kind: ResourceHierarchy
+metadata:
+  name: root-hierarchy
+  namespace: hierarchy`).YNode(),
+			},
+			expected: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: function-input
+data: {apiVersion: cft.dev/v1alpha1, kind: ResourceHierarchy, metadata: {name: root-hierarchy,
+    namespace: hierarchy}}
+`,
+		},
+		{
+			name: "file config",
+			fn:   Function{},
+			configFileContent: `apiVersion: cft.dev/v1alpha1
+kind: ResourceHierarchy
+metadata:
+  name: root-hierarchy
+  namespace: hierarchy`,
+			expected: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: function-input
+data: {apiVersion: cft.dev/v1alpha1, kind: ResourceHierarchy, metadata: {name: root-hierarchy,
+    namespace: hierarchy}}
+`,
+		},
+		{
+			name: "map config",
+			fn: Function{
+				ConfigMap: map[string]string{
+					"foo": "bar",
+				},
+			},
+			expected: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: function-input
+data: {foo: bar}
+`,
+		},
+	}
+
+	for _, c := range cases {
```

Comment:
Use golang subtests here instead of adding the test name to the error messages.

Reply:
done